### PR TITLE
[solution]: zh/medium-diff

### PR DESCRIPTION
**Pull request by SUtengda**
# Challenge [zh/medium-diff]

## Checklist

- [x] Make sure that you are opening the PR with a single solution (one solution per PR)
- [x] The solution must have at least several references for the readers to dive in depth

### CODE REVIEW

File: zh/medium-diff.md [hunk 1]
```diff
@@ -0,0 +1,82 @@
+---
+id: 645
+title: Diff
+lang: en
```
Comment (ghaiklor) [inline zh/medium-diff.md:4]: You need to reflect the language here

```suggestion
lang: zh
```

File: zh/medium-diff.md [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+
+
+让我们从映射类型开始，在这里我们迭代两个对象的属性联合（union）。
+毕竟在计算差值之前，我们需要收集两个对象的所有属性。
+
+```typescript
+type Diff<O, O1> = { [P in keyof O | keyof O1]: never };
+```
+
+When iterating over the properties, we need to check if the property exists on `O` or `O1`.
```
Comment (ghaiklor) [inline zh/medium-diff.md:43]: Missed

**Review by ghaiklor — CHANGES_REQUESTED** [review]

File: zh/medium-diff.md (hunk 1 above)
Comment (SUtengda) [reply zh/medium-diff.md]: Corrected. Thanks

File: zh/medium-diff.md (hunk 2 above)
Comment (SUtengda) [reply zh/medium-diff.md]: done

**Comment by ghaiklor** [comment]
@charleserious is the native to Chinese AF AIU, so we need to get approval of someone like him or himself. In regard to the infra, everything is ok.

**Comment by SUtengda** [comment]
@ghaiklor  It seems all checks are successful for this PR, could you pls merge it ?

**Review by ghaiklor — APPROVED** [review]

**Review by charleserious — COMMENTED** [review]
LGTM

**Pull request merged**